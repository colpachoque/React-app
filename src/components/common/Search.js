import React from 'react';
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';
import RadioGroup from '@material-ui/core/RadioGroup';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Radio from '@material-ui/core/Radio';
import styles from '../header/Header.css';

const Search = (props) => (
    <div className={styles.search}>
        <div>
            {props.label}
        </div>
        <div>
            <TextField
                value={props.value}
                style={{width: '100%'}}
                onChange={({ target }) => props.onChange(target.value)}
            />
        </div>
        <div className={styles.searchBySection}>
            <div className={styles.searchByLabel}>
                SEARCH BY
            </div>
            <div className={styles.radioGroup}>
                <RadioGroup 
                    row
                    value={props.searchBy}
                    onChange={({ target }) => props.onSearchFilterChange(target.value)}>
                    <FormControlLabel
                        value="title"
                        control={<Radio color="primary" />}
                        label="Title"
                        labelPlacement="start"
                    />
                    <FormControlLabel
                        value="genre"
                        control={<Radio color="primary" />}
                        label="Genre"
                        labelPlacement="start"
                    />
                </RadioGroup>
            </div>
            <Button>Search</Button>
        </div>
    </div>
);

export default Search;