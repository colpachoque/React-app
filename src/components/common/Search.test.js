import React from "react";
import Search from "./Search";
import { shallow, mount } from "enzyme";
import toJson from "enzyme-to-json";
import Button from "@material-ui/core/Button";
describe("Search component", () => {
  let movie;

  beforeAll(() => {
    movie = {
      name: "The best movie ever",
      genre: "Drama",
      year: 1994
    };
  });

  it("should be render correctly", () => {
    const component = shallow(
      <Search label={"Tracker"} value={""} searchBy={"title"} />
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it("user text is echoed", () => {
    const component = shallow(
      <Search label={"Tracker"} value={""} searchBy={"title"} />
    );
  });
});
