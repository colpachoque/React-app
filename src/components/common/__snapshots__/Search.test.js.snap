// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search component should be render correctly 1`] = `
<div
  className="search"
>
  <div>
    Tracker
  </div>
  <div>
    <TextField
      onChange={[Function]}
      required={false}
      select={false}
      style={
        Object {
          "width": "100%",
        }
      }
      value=""
      variant="standard"
    />
  </div>
  <div
    className="searchBySection"
  >
    <div
      className="searchByLabel"
    >
      SEARCH BY
    </div>
    <div
      className="radioGroup"
    >
      <RadioGroup
        onChange={[Function]}
        row={true}
        value="title"
      >
        <WithStyles(WithFormControlContext(FormControlLabel))
          control={
            <WithStyles(Radio)
              color="primary"
            />
          }
          label="Title"
          labelPlacement="start"
          value="title"
        />
        <WithStyles(WithFormControlContext(FormControlLabel))
          control={
            <WithStyles(Radio)
              color="primary"
            />
          }
          label="Genre"
          labelPlacement="start"
          value="genre"
        />
      </RadioGroup>
    </div>
    <WithStyles(Button)
      onClick={[Function]}
    >
      Search
    </WithStyles(Button)>
  </div>
</div>
`;
